### PR TITLE
feat: Allow overriding namespace + make default namespace fully qualified

### PR DESCRIPTION
**Pull request by srid**
**PR description**

- The user can now set `namespace` for the processes under each service
- Default namespace is now `${service}.${name}` (previously, just `${name}`)

**Tasks**

- [x] Do this for postgres, to begin with.
- [x] Rest of the services

image

After the user overrides it:

image

After the user overrides the local processes as well:

image

### CODE REVIEW

**Comment by srid** [comment]
The other thing we can do here is to support a special service name, `"default"`. Handy for cases where you run exactly 1 service, so both default namespace and the process name can drop it. Instead of `db-init`, it would just be `postgres-init`.

**Comment by srid** [comment]
To be squash merged.

File: nix/lib.nix [hunk 1]
```diff
@@ -8,25 +8,62 @@
...
+      service = lib.pipe mod [
         builtins.baseNameOf
         (lib.strings.splitString ".")
         builtins.head
       ];
+      serviceModule = { config, name, ... }: {
+        options = {
+          namespace = lib.mkOption {
+            description = ''
+              Namespace for the ${service} service
```
Comment (shivaraj-bh) [inline nix/lib.nix:20]:
```suggestion
              Namespace for the ${service}.${name} service
```

File: nix/lib.nix [hunk 2]
```diff
@@ -8,25 +8,62 @@
...
+            '';
+            default = "${service}.${name}";
+            type = lib.types.str;
+          };
+          outputs.defaultProcessSettings = lib.mkOption {
+            type = lib.types.deferredModule;
+            internal = true;
+            readOnly = true;
+            description = ''
+              Default settings for all processes under the ${service} service
```
Comment (shivaraj-bh) [inline nix/lib.nix:30]:
```suggestion
              Default settings for all processes under the ${service}.${name} service
```

File: nix/postgres/default.nix [hunk 3]
```diff
@@ -301,7 +300,7 @@ in
               in
               {
                 command = setupScript;
-                namespace = name;
+                inherit (config) namespace;
```
Comment (shivaraj-bh) [inline nix/postgres/default.nix:303]:
```suggestion
```

File: nix/postgres/default.nix [hunk 4]
```diff
@@ -329,6 +328,7 @@ in
                 ] ++ (lib.optional (config.superuser != null) "-U ${config.superuser}");
               in
               {
+                inherit (config) namespace;
```
Comment (shivaraj-bh) [inline nix/postgres/default.nix:331]:
```suggestion
```

**Review by shivaraj-bh — CHANGES_REQUESTED** [review]

File: nix/lib.nix (hunk 2 above)
Comment (srid) [reply nix/lib.nix:30]: If `name` is dynamic, how will the docs render?

File: nix/lib.nix (hunk 1 above)
Comment (srid) [reply nix/lib.nix:20]: (See below)

File: nix/lib.nix (hunk 2 above)
Comment (shivaraj-bh) [reply nix/lib.nix:30]: My bad, I missed that.

**Pull request merged**